pkg/engine/executor/pipeline_test.go

```diff
@@ -0,0 +1,494 @@
+package executor
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+	"github.com/stretchr/testify/require"
+)
+
+// CSVToArrow converts a CSV string to an Arrow record based on the provided schema.
+// It supports int64, float64, and boolean types for now.
+func CSVToArrow(fields []arrow.Field, csvData string) (arrow.Record, error) {
+	// Create schema
+	schema := arrow.NewSchema(fields, nil)
+
+	// Prepare memory allocator
+	mem := memory.NewGoAllocator()
```

Comment:
We should permit passing in an allocator here; that way we can pass in a checked allocator and make sure that node implementations properly release records when they're done with them.